cg/services/deliver_files/deliver_files_service/deliver_files_service_factory.py

```diff
@@ -122,18 +102,23 @@ def _validate_delivery_type(delivery_type: DataDelivery):
         ]:
             return
         raise DeliveryTypeNotSupported(
-            f"Delivery type {delivery_type} is not supported. Supported delivery types are {DataDelivery.FASTQ}, {DataDelivery.ANALYSIS_FILES}, {DataDelivery.FASTQ_ANALYSIS}, {DataDelivery.BAM}."
+            f"Delivery type {delivery_type} is not supported. Supported delivery types are"
+            f" {DataDelivery.FASTQ}, {DataDelivery.ANALYSIS_FILES},"
+            f" {DataDelivery.FASTQ_ANALYSIS}, {DataDelivery.BAM}."
         )
 
     def build_delivery_service(
-        self, workflow: Workflow, delivery_type: DataDelivery
+        self, case: Case, delivery_type: DataDelivery | None = None
     ) -> DeliverFilesService:
-        """Build a delivery service based on the workflow and delivery type."""
-        delivery_type: DataDelivery = self._sanitise_delivery_type(delivery_type)
+        """Build a delivery service based on a case."""
+        delivery_type: DataDelivery = self._sanitise_delivery_type(
+            delivery_type if delivery_type else case.data_delivery
+        )
         self._validate_delivery_type(delivery_type)
```

Comment:
"sanitise" and "validate" seem to share some responsibility? If something has been sanitised, it should not need to be validated

Reply:
out of scope

Reply:
Input sanitisation: cleaning or transforming input
Input validation: making sure input is semantically correct

Related but not the same